Week-6/mapSetEcommerce/script.js

```diff
@@ -0,0 +1,38 @@
+const productViews = new WeakMap();
```

Comment:
Using WeakMap and WeakSet is appropriate for non-persistent data, but be aware that these structures are not iterable.